test/arbitration_spec.js

```diff
@@ -0,0 +1,305 @@
+
+  const tradeAmount = 100;
+  const feeAmount = Math.round(tradeAmount * (feePercent / 100));
+
+  // util
+  let expirationTime = parseInt((new Date()).getTime() / 1000, 10) + 1000;
+  const expireTransaction = async() => {
+    await TestUtils.increaseTime(1001);
+    expirationTime += 1000;
+  };
+
+  let receipt, escrowId, escrowTokenId, _offerId, ethOfferId, tokenOfferId, hash, signature, nonce;
+
+  this.timeout(0);
+
+  before(async () => {
+    await SNT.methods.generateTokens(accounts[0], 1000).send();
+    const encodedCall = SellerLicense.methods.buy().encodeABI();
+    await SNT.methods.approveAndCall(SellerLicense.options.address, 10, encodedCall).send({from: accounts[0]});
+
+    // Register arbitrators
+    await SNT.methods.generateTokens(arbitrator, 1000).send();
+    await SNT.methods.generateTokens(arbitrator2, 1000).send();
+
+    const encodedCall2 = ArbitrationLicense.methods.buy().encodeABI();
+    await SNT.methods.approveAndCall(ArbitrationLicense.options.address, 10, encodedCall2).send({from: arbitrator});
+    await SNT.methods.approveAndCall(ArbitrationLicense.options.address, 10, encodedCall2).send({from: arbitrator2});
+
+    receipt  = await MetadataStore.methods.addOffer(TestUtils.zeroAddress, "0x00", "London", "USD", "Iuri", [0], 1, arbitrator).send({from: accounts[0]});
+    ethOfferId = receipt.events.OfferAdded.returnValues.offerId;
+    receipt  = await MetadataStore.methods.addOffer(StandardToken.options.address, "0x00", "London", "USD", "Iuri", [0], 1, arbitrator).send({from: accounts[0]});
+    tokenOfferId = receipt.events.OfferAdded.returnValues.offerId;
+
+     const abiEncode = Escrow.methods.init(
+      EscrowRelay.options.address,
+      SellerLicense.options.address,
+      Arbitrations.options.address,
+      MetadataStore.options.address,
+      "0x0000000000000000000000000000000000000002",
+      FEE_MILLI_PERCENT
+    ).encodeABI();
+
+    await EscrowManagement.methods.setTemplate(Escrow.options.address, abiEncode).send();
+  });
+  
+  describe("Transaction arbitration case", async() => {
+    beforeEach(async() => {
+      // Create
+      hash = await MetadataStore.methods.getDataHash("U", "0x00").call({from: accounts[1]});
+      signature = await web3.eth.sign(hash, accounts[1]);
+      nonce = await MetadataStore.methods.user_nonce(accounts[1]).call();
+      receipt = await EscrowManagement.methods.create(ethOfferId, tradeAmount, FIAT, 140, "0x00", "L", "U", nonce, signature).send({from: accounts[1]});
+      Escrow.options.address = receipt.events.InstanceCreated.returnValues.instance;
+      // Fund
+      receipt = await Escrow.methods.fund(tradeAmount, expirationTime).send({from: accounts[0], value: tradeAmount + feeAmount});
+    });
+
+    it("should allow a buyer to open a case", async() => {
+      await Escrow.methods.pay().send({from: accounts[1]});
+
+      Escrow.options.jsonInterface.push(Arbitrations.options.jsonInterface.find(x => x.name === "ArbitrationRequired"));
```

Comment:
Is this because the Arbitration methods are not present in the Escrow json interface?

Reply:
I did it because `Escrow` will call `Arbitrations.openCase`, and the `ArbitrationRequired` event will be emitted there, but since `Escrow` does not know the details of that event, I added it to its interface in order to be able to decode that log.

Maybe we should add a configuration attribute to the contracts config in Embark to simplify this procedure? something like:
```
contracts {
     "Arbitrations":  { ... },
     "Escrow": {
        ...,
        "inheritEventsFrom": ["Arbitrations"]
     }
}
```
what do you think?

Reply:
Yeah, to be fair, it would be even cooler if Embark detected inheritance and put all the events of the parents in the json interface, but `inheritEventsFrom` could be a good a good alternative